fix(rpc): preserve nested bundle structure in mev_simBundle logs

meetrick opened the pull request:
Fixes log collection in `mev_simBundle` so nested bundles preserve their hierarchical structure.

The implementation now records logs in execution order and rebuilds the `SimBundleLogs` tree iteratively from the original bundle shape. This avoids misattributing logs when the same signed transaction appears multiple times in a bundle and adds regression tests for duplicate transactions and mismatched log reconstruction.

meetrick committed:
fix: preserve nested bundle structure when collecting logs

- Add build_bundle_logs() to recursively build SimBundleLogs tree
- Add raw_tx_hash field to FlattenedBundleItem for consistent hash matching
- Use HashMap to map transaction hashes to logs
- Add unit tests for nested bundle log collection

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

meetrick committed:
Code clean and modify test code

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

mattsse reviewed (changes requested):
thanks, all of this makes sense

but I believe we can also populate the logs if use smth like a stack of logs instead because I'd like to avoid having recursive calls here

meetrick commented on crates/rpc/rpc/src/eth/sim_bundle.rs:
```diff
@@ -330,15 +374,17 @@ where
-                        let sim_bundle_logs =
-                            SimBundleLogs { tx_logs: Some(tx_logs), bundle_logs: None };
-                        body_logs.push(sim_bundle_logs);
+                        tx_logs_map.insert(tx_hash, tx_logs);
```
got it! thanks

meetrick committed:
refactor: replace recursive build_bundle_logs with iterative approach

- Replace recursive build_bundle_logs() with stack-based iteration
- Add bundle_path to FlattenedBundleItem for tracking item position
- Revert formatting changes per review feedback
- Update tests to use iterative approach

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

meetrick committed:
Deleted unnecessary comment

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

meetrick committed:
fixed apply code formatting and fix clippy warnings

- Fix formatting issues detected by cargo fmt
- Remove redundant clone() calls in test code (clippy::redundant-clone)

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

meetrick committed:
Merge branch 'main' into reth/sim_bundle_todo

meetrick commented:
@mattsse Could you review again?

meetrick committed:
Merge branch 'main' into reth/sim_bundle_todo

meetrick committed:
Merge branch 'main' into reth/sim_bundle_todo

meetrick committed:
chore(rpc): remove unused bundle_path field from FlattenedBundleItem

The bundle_path field was added but never used. The nested bundle
log structure is reconstructed by re-traversing the original bundle
and matching transactions via raw_tx_hash, making bundle_path redundant.

Signed-off-by: Hwangjae Lee <meetrick@gmail.com>

meetrick commented on crates/rpc/rpc/src/eth/sim_bundle.rs:
```diff
@@ -59,6 +62,9 @@ pub struct FlattenedBundleItem<T> {
     pub refund_configs: Option<Vec<RefundConfig>>,
+    /// Path from root bundle to this item, represented as indices at each nesting level.
+    /// For example, `[1, 0]` means `bundle.bundle_body[1].bundle.bundle_body[0]`.
+    pub bundle_path: Vec<usize>,
```
@mattsse  I removed bundle_path. You commented because it was unused and should be cleaned up, right?

decofe committed:
chore: resolve merge conflict with main

Take block_hash/block_number/block_timestamp from main while keeping
raw_tx_hash for transaction_hash from the PR branch.

Co-Authored-By: Matthias Seitz <19890894+mattsse@users.noreply.github.com>
Amp-Thread-ID: https://ampcode.com/threads/T-019d7710-7801-7100-9064-15e8f76472f2

mattsse committed:
fix(rpc): rebuild mev_simBundle logs by execution order

mattsse committed:
docs(rpc): restore refund calculation comment

mattsse committed:
test(rpc): cover additional sim bundle log shapes
